library/doxygen/source/conf.py

```diff
@@ -0,0 +1,85 @@
+# Configuration file for the Sphinx documentation builder.
+#
+# This file only contains a selection of the most common options. For a full
+# list see the documentation:
+# https://www.sphinx-doc.org/en/master/usage/configuration.html
+
+# -- Path setup --------------------------------------------------------------
+
+# If extensions (or modules to document with autodoc) are in another directory,
+# add these directories to sys.path here. If the directory is relative to the
+# documentation root, use os.path.abspath to make it absolute, like shown here.
+#
+# import os
+# import sys
+# sys.path.insert(0, os.path.abspath('.'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'libf3d'
+copyright = '2022, Michael Migliore and Mathieu Westphal'
+author = 'Michael Migliore and Mathieu Westphal'
+
+# The full version, including alpha/beta/rc tags
+release = '1.3.1'
+
+
+# -- General configuration ---------------------------------------------------
+
+# Add any Sphinx extension module names here, as strings. They can be
+# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
+# ones.
+extensions = [
+    'breathe',
+    'exhale'
+]
+
+# Add any paths that contain templates here, relative to this directory.
+templates_path = ['_templates']
+
+# List of patterns, relative to source directory, that match files and
+# directories to ignore when looking for source files.
+# This pattern also affects html_static_path and html_extra_path.
+exclude_patterns = ['_build', 'Thumbs.db', '.DS_Store']
+
+
+# -- Options for HTML output -------------------------------------------------
+
+# The theme to use for HTML and HTML Help pages.  See the documentation for
+# a list of builtin themes.
+#
+html_theme = 'sphinx_book_theme'
+
+# Add any paths that contain custom static files (such as style sheets) here,
+# relative to this directory. They are copied after the builtin static files,
+# so a file named "default.css" will overwrite the builtin "default.css".
+html_static_path = ['_static']
+
+# Setup the breathe extension
+breathe_projects = {
+    "libf3d": "./_doxygen/xml"
+}
+breathe_default_project = "libf3d"
+
+# Setup the exhale extension
+exhale_args = {
+    # These arguments are required
+    "containmentFolder":     "./api",
+    "rootFileName":          "library_root.rst",
+    "doxygenStripFromPath":  "..",
+    # Heavily encouraged optional argument (see docs)
+    "rootFileTitle":         "Library API",
+    # Suggested optional arguments
+    "createTreeView":        True,
+    # TIP: if using the sphinx-bootstrap-theme, you need
+    # "treeViewIsBootstrap": True,
+    "exhaleExecutesDoxygen": True,
+    "exhaleDoxygenStdin":    "INPUT = ../../public\nDISTRIBUTE_GROUP_DOC = YES\nEXCLUDE_SYMBOLS = std"
+}
+
+# Tell sphinx what the primary language being documented is.
+primary_domain = 'cpp'
```

Comment:
can we document the CMake files too? `f3dEmbed.cmake` is already in the SDK, and `f3dPlugin.cmake` will be included soon too.

Reply:
no idea if doxygen supports that, ill take a look.

Reply:
Doxygen do not support documenting cmake files: 

https://www.doxygen.nl/manual/starting.html#default_file_extension_mapping

I tried to force parsing on f3dPlugin.cmake anyway and it fails with:

```
/home/glow/dev/f3d/f3d/src/library/doxygen/api/program_listing_file__home_glow_dev_f3d_f3d_src_cmake_f3dPlugin.h.rst:11: WARNING: Could not lex literal_block as "cpp". Highlighting skipped.
```